game-core/src/test/java/games/strategy/engine/data/CompositeChangeTest.java

```diff
@@ -0,0 +1,41 @@
+package games.strategy.engine.data;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class CompositeChangeTest {
+
+  @Test
+  void flattenChanges() {
```

Comment:
Couple more test cases to consider having:
- trivial case where there is a single composite change and the input is the same as output
- deeply nested (more than 2) singleton change

Reply:
The existing test case already is testing deeply nested changes.  It goes at least 3 deep.

Reply:
Indeed, I'm drawing a slight distinction:
- deeply nested with N+1 changes
- deeply nested with just 1 change

In the latter case one would expect that to be flattened into a single change.

Reply:
@DanVanAtta , I see what you mean.  I've added a test for the later case.